shared/cancel/canceller.go

```diff
@@ -0,0 +1,19 @@
+package cancel
+
+import "context"
+
+// Canceller is a simple wrapper for a cancellable context which makes the associated context.CancelFunc more easily
+// accessible.
+type Canceller struct {
+	context.Context
+	CancelFunc func()
```

Comment:
@markylaing sorry you misunderstood me, I meant like this `Cancel context.CancelFunc`

Reply:
Ohhh ok sure